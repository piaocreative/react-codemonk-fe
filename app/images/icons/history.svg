<svg width="18" height="18" viewBox="0 0 18 18" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M1 9C1 13.4186 4.58141 17 9 17C13.4186 17 17 13.4186 17 9C17 4.58141 13.4186 1 9 1C6.25007 1 3.8244 2.38717 2.3845 4.5" stroke="#04004E" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round"/>
<path d="M11.9679 11.5448L8.70732 9.5997V5.4077" stroke="#04004E" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round"/>
<path d="M2.10001 2L2.10001 4.82843H4.92843" stroke="#04004E" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round"/>
</svg>
