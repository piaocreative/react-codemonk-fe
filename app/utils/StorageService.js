import base from 'base-64';

/**
 * This class used for Store value on localStorage of browser.
 * @author Innovify
 */

const StorageService = {
  /**
   * check value exists on localStorage of browser.
   * @author Innovify
   * @Developer Ankit Gujarati
   */
  exists(key) {
    return this.get(key);
  },
  /**
   * set value from localStorage of browser.
   * @author Innovify
   * @Developer Ankit Gujarati
   */
  set(key, value, opts = {}) {
    let storedValue = value;
    if (opts.stringify) storedValue = JSON.stringify(storedValue);
    if (opts.hash) storedValue = base.encode(storedValue);
    try {
      if (opts.keepMe) {
        const d = new Date();
        d.setTime(d.getTime() + 30 * 24 * 60 * 60 * 1000);
        const expires = `expires=${d.toUTCString()}`;
        document.cookie = `${key}=${JSON.stringify({
          opts,
          body: storedValue,
        })};${expires};path=/`;
      } else {
        document.cookie = `${key}=${JSON.stringify({
          opts,
          body: storedValue,
        })};path=/`;
      }
    } catch (err) {
      throw err;
    }
  },
  /**
   * get value from localStorage of browser.
   * @author Innovify
   */
  get(key) {
    // if (!this.exists(key)) return false;
    const name = `${key}=`;
    const decodedCookie = decodeURIComponent(document.cookie);
    const ca = decodedCookie.split(';');
    // eslint-disable-next-line no-plusplus
    for (let i = 0; i < ca.length; i++) {
      let c = ca[i];
      while (c.charAt(0) === ' ') {
        c = c.substring(1);
      }
      if (c.indexOf(name) === 0) {
        const data = JSON.parse(c.substring(name.length, c.length));
        let { body } = data;
        if (data.opts && data.opts.hash) body = base.decode(body);
        if (data.opts && data.opts.stringify) body = JSON.parse(body);
        return body;
      }
    }
    return false;
    /* const data = JSON.parse(localStorage.getItem(key));
    let { body } = data;
    if (data.opts && data.opts.hash) body = base.decode(body);
    if (data.opts && data.opts.stringify) body = JSON.parse(body);
    return body; */
  },
  /**
   * delete value from localStorage of browser.
   * @author Innovify
   */
  delete(key, opts = {}) {
    const d = new Date();
    d.setTime(d.getTime() + 0);
    const expires = `expires=${d.toUTCString()}`;
    document.cookie = `${key}=${JSON.stringify({
      opts,
      body: '',
    })};${expires};path=/`;
    // localStorage.removeItem(key);
  },
  /**
   * clear localStorage of browser.
   * @author Innovify
   */
  clear() {
    this.delete('signupStep');
    this.delete('apiSignupStep');
    this.delete('userVersion');
    this.delete('user');
    this.delete('recentSearch');
    this.delete('jwtToken');
    this.delete('location');
    this.delete('unAuthorize');
    this.delete('userPhoneCountry');
    this.delete('registerType');
    this.delete('userEmail');
    this.delete('userType');
    this.delete('agencyLogo');
    this.delete('pageNumber');
    this.delete('userPhoneNumber');
    this.delete('profilePicture');
    this.delete('firstName');
    this.delete('userPhone');
    this.delete('Notification');
    this.delete('NotificationType');
    this.delete('talentProjectPageNumber');
    this.delete('talentProjectFilter');
    this.delete('clientProjectPageNumber');
    this.delete('clientProjectStatus');
    this.delete('isPaymentSkipped');
    this.delete('agencyMyTeamPageNumber');
    this.delete('adminAgencyPageNumber');
    this.delete('adminAgencyFilterObject');
    this.delete('adminBriefDetailID');
    this.delete('clientsFilterObject');
    this.delete('clientsPageNumber');
    this.delete('interviewFilterObject');
    this.delete('interviewPageNumber');
    this.delete('adminProjectDetailsBriefPageNumber');
    this.delete('projectFilterObject');
    this.delete('projectPageNumber');
    this.delete('talentFilterObject');
    this.delete('talentPageNumber');
    this.delete('agencyQuotePage');
    this.delete('selectedRoleType');
    this.delete('filterObject');
    this.delete('proxyType');
    this.delete('proxyToken');
    this.delete('talentBriefPage');
    this.delete('talentBriefFilter');
    this.delete('clientBriefDetailsPageNumber');
    this.delete('clientBriefPage');
    this.delete('clientBriefFilter');
    this.delete('clientID');
    this.delete('adminTalentSearch');
    this.delete('adminAgencySearch');
    this.delete('adminClientSearch');
    this.delete('isLoginViaAdmin');
    this.delete('userBillingDetailsObj');
  },
};

export default StorageService;
