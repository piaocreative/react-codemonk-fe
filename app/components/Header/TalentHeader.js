import React from 'react';
import { userExists, getUserRegisterType } from 'utils/Helper';
import { Link } from 'react-router-dom';
import { FormattedMessage } from 'react-intl';
import get from 'lodash/get';
import { Dropdown, DropdownItem } from 'reactstrap';
import { defaultProps, propTypes } from 'containers/proptypes';
import LogoutLink from 'components/LogoutLink';
import SVG from 'react-inlinesvg';
import StorageService from 'utils/StorageService';
import Emitter from 'utils/emitter';
import { userIcon, settingIcon, logoutIcon, backArrowIcon, helpIcon } from 'containers/App/constants';
import { handleBackToAdmin } from 'containers/Admin/ProxyLogin/utils';
import Notifications from 'containers/Notifications';
import {
  ProfileImg,
  HeaderNav,
  FixedNavbar,
  NavbarContainer,
  UserProfileToggle,
  UserProfileMenu,
  UserImg,
  SecondaryNav,
  HamburgerToggle,
} from './header-style';
import TalentLogo from '../Brand/TalentLogo';
import messages from './messages';

export class TalentHeader extends React.Component {
  constructor(props) {
    super(props);
    const isSignupStep = StorageService.get('signupStep');
    const apiSignupStep = Number(StorageService.get('apiSignupStep'));
    const profilePicture = StorageService.get('profilePicture');
    const agencyLogo = StorageService.get('agencyLogo');
    this.state = {
      dropdownOpen: false,
      isSignIn: userExists(),
      isOpen: false,
      roleType: getUserRegisterType(),
      isSignupStep,
      apiSignupStep,
      profilePicture,
      agencyLogo,
    };
  }

  setConstructor = () => {
    const proxyType = StorageService.get('proxyType');
    const isSignupStep = StorageService.get('signupStep');
    const profilePicture = StorageService.get('profilePicture');
    const agencyLogo = StorageService.get('agencyLogo');
    const stateData = {
      dropdownOpen: false,
      isSignIn: userExists(),
      isOpen: false,
      roleType: getUserRegisterType(),
      isSignupStep,
      profilePicture,
      agencyLogo,
      proxyType,
      badgeConfiguration: {},
    };
    this.setState({ ...stateData });
  };

  componentDidMount() {
    this.setConstructor();

    Emitter.on('userTypeSelected', userTypeSelected => {
      if (userTypeSelected) {
        const profilePicture = StorageService.get('profilePicture');
        const agencyLogo = StorageService.get('agencyLogo');
        const roleType = getUserRegisterType();
        this.setState({
          roleType,
          profilePicture,
          agencyLogo,
        });
      }
    });
    Emitter.on('profilePicture', profilePicture => {
      let url = '';
      if (profilePicture) {
        url = `${profilePicture}?_t=${new Date().getTime()}`;
      }
      this.setState({
        profilePicture: url,
      });
      StorageService.set('profilePicture', url, { hash: true });
    });
    Emitter.on('agencyLogo', agencyLogo => {
      let url = '';
      if (agencyLogo) {
        url = `${agencyLogo}?_t=${new Date().getTime()}`;
      }
      this.setState({
        agencyLogo: url,
      });
      StorageService.set('agencyLogo', url, { hash: true });
    });

    Emitter.on('proxyLoginAgency', proxyLoginAgency => {
      if (proxyLoginAgency) {
        this.setConstructor();
      }
    });

    Emitter.on('badgeConfigurationUpdated', badgeConfigurationUpdated => {
      this.setState({ badgeConfiguration: badgeConfigurationUpdated });
    });
  }

  componentWillUnmount() {
    Emitter.off('userTypeSelected');
    Emitter.off('profilePicture');
    Emitter.off('agencyLogo');
    Emitter.off('proxyLoginAgency');
    Emitter.off('badgeConfigurationUpdated');
  }

  /**
   * call dropdownToggle on tab change
   * @author Innovify
   */
  dropdownToggle = () => {
    this.setState(prevState => ({
      dropdownOpen: !prevState.dropdownOpen,
    }));
  };

  menuToggle = () => {
    this.setState(prevState => ({
      isOpen: !prevState.isOpen,
    }));
  };

  render() {
    const {
      isSignIn,
      dropdownOpen,
      isSignupStep,
      profilePicture,
      agencyLogo,
      roleType,
      proxyType,
      badgeConfiguration,
      isOpen,
      apiSignupStep,
    } = this.state;
    const userVersion = StorageService.get('userVersion');

    let talentOnBoard = false;
    let talentUser = false;
    let agencyUser = false;
    let agencyOnBoard = false;

    if (roleType === 'talent') {
      talentUser = true;
      talentOnBoard = (apiSignupStep >= 6 && userVersion !== 'v2') || (apiSignupStep === 7 && userVersion === 'v2');
    }
    if (roleType === 'talent_agency') {
      talentUser = true;
      talentOnBoard = (apiSignupStep >= 5 && userVersion !== 'v2') || (apiSignupStep === 5 && userVersion === 'v2');
    }
    if (roleType === 'agency') {
      agencyUser = true;
      agencyOnBoard = isSignupStep === '7';
    }
    return (
      <HeaderNav className={isOpen && 'sidebarOpen'}>
        <FixedNavbar>
          <NavbarContainer fluid>
            {this.getBrandLogo(isSignIn, talentOnBoard, agencyOnBoard)}
            {isSignIn && (talentUser || agencyUser) && (
              <div className="d-flex align-items-center">
                <SecondaryNav>
                  <Link to="/dashboard" className="nav-link d-none">
                    <SVG src={helpIcon} />
                  </Link>
                  {(talentOnBoard || agencyOnBoard) && <Notifications newNotification={get(badgeConfiguration, 'newNotification')} />}
                </SecondaryNav>
                <Dropdown isOpen={dropdownOpen} toggle={this.dropdownToggle}>
                  <UserProfileToggle>
                    {talentUser && profilePicture && <UserImg src={profilePicture} alt="user-profile" />}
                    {talentUser && !profilePicture && (
                      <ProfileImg>
                        <SVG src={userIcon} />
                      </ProfileImg>
                    )}
                    {agencyUser && agencyLogo && <UserImg src={agencyLogo} alt="user-profile" />}
                    {agencyUser && !agencyLogo && (
                      <ProfileImg>
                        <SVG src={userIcon} />
                      </ProfileImg>
                    )}
                  </UserProfileToggle>
                  <UserProfileMenu right>
                    <div className="inner-block">
                      {talentUser && talentOnBoard && (
                        <DropdownItem tag={Link} to="/talent/my-profile">
                          <SVG src={userIcon} />
                          <FormattedMessage {...messages.menuItemMyProfile} />
                        </DropdownItem>
                      )}
                      {agencyUser && agencyOnBoard && (
                        <DropdownItem tag={Link} to="/agency/agency-profile">
                          <SVG src={userIcon} />
                          <FormattedMessage {...messages.menuItemMyProfile} />
                        </DropdownItem>
                      )}
                      {talentUser && talentOnBoard && (
                        <DropdownItem tag={Link} to="/talent/account-settings">
                          <SVG src={settingIcon} />
                          <FormattedMessage {...messages.menuItemAccountSetting} />
                        </DropdownItem>
                      )}

                      {agencyUser && agencyOnBoard && (
                        <DropdownItem tag={Link} to="/agency/agency-account-settings">
                          <SVG src={settingIcon} />
                          <FormattedMessage {...messages.menuItemAccountSetting} />
                        </DropdownItem>
                      )}
                        <LogoutLink>
                          <SVG src={logoutIcon} />
                          <FormattedMessage {...messages.menuItemLogout} />
                        </LogoutLink>

                      {/* proxyLogin Return to admin */}
                      {proxyType && (
                        <DropdownItem onClick={handleBackToAdmin}>
                          <SVG src={backArrowIcon} />
                          <FormattedMessage {...messages.menuItemBackToAdmin} />
                        </DropdownItem>
                      )}
                    </div>
                  </UserProfileMenu>
                </Dropdown>
              </div>
            )}
          </NavbarContainer>
        </FixedNavbar>
      </HeaderNav>
    );
  }

  getBrandLogo(isSignIn, talentOnBoard, agencyOnBoard) {
    return (
      <div className="d-flex align-items-center navbar-expand-md">
        <TalentLogo />
        {isSignIn && (talentOnBoard || agencyOnBoard) && <HamburgerToggle onClick={this.menuToggle} />}
      </div>
    );
  }
}

TalentHeader.defaultProps = defaultProps;
TalentHeader.propTypes = propTypes;

export default TalentHeader;
