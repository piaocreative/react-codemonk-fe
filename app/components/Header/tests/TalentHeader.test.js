import React from 'react';
import { shallow } from 'enzyme';
import { render } from 'react-testing-library';
import { Provider } from 'react-redux';
import { IntlProvider } from 'react-intl';
import { ConnectedRouter } from 'connected-react-router/immutable';
import { createMemoryHistory } from 'history';
import StorageService from 'utils/StorageService';
import Emitter from 'utils/emitter';

import TalentHeader from '../TalentHeader';
import configureStore from '../../../configureStore';

describe('<TalentHeader />', () => {
  const history = createMemoryHistory();
  const store = configureStore({}, history);

  it('should render a div', () => {
    const { container } = render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <ConnectedRouter history={history}>
            <TalentHeader />
          </ConnectedRouter>
        </IntlProvider>
      </Provider>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('Test emitter functions', () => {
  test('Testing if toggle with userTypeSelected = true', () => {
    const component = shallow(<TalentHeader />);
    StorageService.set('profilePicture', 'picture');
    StorageService.set('userType', '1');
    StorageService.set('registerType', 'freelancer');
    Emitter.emit('userTypeSelected', true);
    expect(component.state().profilePicture).toEqual('picture');
  });

  test('Testing if toggle with userTypeSelected = false', () => {
    const component = shallow(<TalentHeader />);
    StorageService.set('profilePicture', 'picture');
    StorageService.set('userType', '1');
    StorageService.set('registerType', 'freelancer');
    Emitter.emit('userTypeSelected', false);
    expect(component.state().roleType).toEqual('talent');
  });

  // profilePicture Emitter
  test('Testing if toggle with profilePicture = true', () => {
    const component = shallow(<TalentHeader />);
    Emitter.emit('profilePicture', 'image');
    const newImage = StorageService.get('profilePicture');
    expect(component.state().profilePicture).toEqual(newImage);
  });

  test('Testing if toggle with profilePicture = false', () => {
    const component = shallow(<TalentHeader />);
    Emitter.emit('profilePicture', '');
    expect(component.state().profilePicture).toEqual('');
  });

  // agencyLogo Emitter
  test('Testing if toggle with agencyLogo = true', () => {
    const component = shallow(<TalentHeader />);
    Emitter.emit('agencyLogo', 'img');
    const newImage = StorageService.get('agencyLogo');
    expect(component.state().agencyLogo).toEqual(newImage);
  });

  test('Testing if toggle with agencyLogo = false', () => {
    const component = shallow(<TalentHeader />);
    Emitter.emit('agencyLogo', '');
    expect(component.state().agencyLogo).toEqual('');
  });

  // proxyLoginAgency Emitter
  test('Testing if toggle with proxyLoginAgency = true', () => {
    const component = shallow(<TalentHeader />);
    StorageService.set('proxyType', 'agency', { hash: true });
    Emitter.emit('proxyLoginAgency', true);
    expect(component.state().proxyType).toEqual('agency');
  });

  test('Testing if toggle with proxyLoginAgency = false', () => {
    const component = shallow(<TalentHeader />);
    StorageService.set('proxyType', 'admin', { hash: true });
    Emitter.emit('proxyLoginAgency', false);
    expect(component.state().proxyType).toEqual('agency');
  });

  // componentWillUnmount
  test('Testing if emitter off on component unmount', () => {
    const component = shallow(<TalentHeader />);
    jest.spyOn(component.instance(), 'componentWillUnmount');
    component.instance().componentWillUnmount();
    expect(component.instance().componentWillUnmount).toHaveBeenCalledTimes(1);
  });
});

describe('Test functions', () => {
  const getWrapper = () => shallow(<TalentHeader />);
  const getInstance = () => getWrapper().instance();

  test('Testing if dropdownToggle', () => {
    const dropdownToggle = jest.spyOn(getInstance(), 'dropdownToggle');
    dropdownToggle();
    expect(dropdownToggle).toHaveBeenCalledTimes(1);
  });

  test('Testing if menuToggle', () => {
    const menuToggle = jest.spyOn(getInstance(), 'menuToggle');
    menuToggle();
    expect(menuToggle).toHaveBeenCalledTimes(1);
  });
});
