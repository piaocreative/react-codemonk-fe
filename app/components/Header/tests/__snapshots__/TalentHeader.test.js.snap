// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TalentHeader /> should render a div 1`] = `
<div
  class="header-style__HeaderNav-sc-196epht-5 bxBvEb"
>
  <nav
    class="header-style__FixedNavbar-sc-196epht-6 ecmbTL navbar"
  >
    <div
      class="header-style__NavbarContainer-sc-196epht-12 gqcBPh container-fluid"
    >
      <div
        class="d-flex align-items-center navbar-expand-md"
      >
        <a
          class="brand-style__NavBrand-kuuhq1-0 eghcJN"
          href="/"
          title="CodeMonk"
        >
          <img
            alt="CodeMonk"
            src="IMAGE_MOCK"
          />
        </a>
      </div>
    </div>
  </nav>
</div>
`;
